src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -283,7 +283,14 @@ public static CloseableHttpResponse execute(
         // if an SSL issue occurs like an SSLHandshakeException then fail
         // immediately and stop retrying the requests
 
-        throw new SnowflakeSQLLoggedException(null, ErrorCode.NETWORK_ERROR, ex, ex.getMessage());
+        String formattedMsg =
+            ex.getMessage()
+                + "\n"
+                + "Verify that the hostnames and portnumbers in SYSTEM$ALLOWLIST are added to your firewall's allowed list.\n"
+                + "To troubleshoot your connection further, you can refer to this article:\n"
+                + "https://docs.snowflake.com/en/user-guide/client-connectivity-troubleshooting/overview";
```

Comment:
Don't we want to point to JDBC errors? https://docs.snowflake.com/en/user-guide/client-connectivity-troubleshooting/error-messages#jdbc-errors

Reply:
I would prefer to point to the root of the troubleshooting guide where are more information about the errors

Reply:
We agreed that we can go with the root link